pinot-plugins/pinot-stream-ingestion/pinot-pulsar/src/main/java/org/apache/pinot/plugin/stream/pulsar/PulsarPartitionLevelConsumer.java

```diff
@@ -93,6 +96,10 @@ public PulsarMessageBatch fetchMessages(MessageId startMessageId, MessageId endM
           }
         }
         messagesList.add(nextMessage);
+
```

Comment:
Isnt it better to catch interruptedException, and return the messages fetched so far?

Reply:
That's what I wanted to do but Java is not allowing that saying, the interrupted exception is never thrown in this block.

Reply:
grr...